tests/test_external_tests.py

```diff
@@ -1,90 +1,121 @@
+import os
```

Comment:
(No changed needed, but I wanted to share some resources with you).

This article gives a great overview of Behavior Driven Development (BDD) topics that can be applied to testing: https://pythontest.com/strategy/given-when-then-2/

BDD is also heavily used by automation engineers when writing automated end-to-end testing in a number of frameworks, so it's likely you'll run into this eventually.

I find thinking in a GIVEN-WHEN-THEN structure for my tests gives me clearer focus that I am testing a singular thing, I know what my assumptions are, and I know the actions I am taking.

I started to use this structure in the Synapse Python Client:

1. https://github.com/Sage-Bionetworks/synapsePythonClient/commit/0d1721fffd74231e477d8c15dad6a98e0988df53#diff-5056ccf1a22d1e69ce5aeea47808fc00ebe8173df1b70a9eb6cd02d2d38cad51R505
2. https://github.com/Sage-Bionetworks/synapsePythonClient/commit/c6223b56e2b1963b880af33a8c19c9c585875a2b#diff-eb75db4e0b13e5e4845036998dcf3c8c8c5509fb0560b21f5e7d1d21ec448d7bR356

Reply:
This is great food for thought and I will look to incorporate these concepts in future work!